<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>100</x>
    <y>100</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Stock Data Analysis</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="controlPanel">
      <property name="title">
       <string>Controls</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QGroupBox" name="searchGroup">
         <property name="title">
          <string>Search and Modify</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Symbol:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="symbolInput">
            <property name="placeholderText">
             <string>Enter Symbol</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="searchButton">
            <property name="text">
             <string>Search &amp; Reduce Price by 1/2</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="addGroup">
         <property name="title">
          <string>Add New Data</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>New Data:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="newSymbol">
            <property name="placeholderText">
             <string>Symbol</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="newPrice">
            <property name="placeholderText">
             <string>Price</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="newPE">
            <property name="placeholderText">
             <string>PE</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="newGroup">
            <property name="placeholderText">
             <string>Group</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addButton">
            <property name="text">
             <string>Add Data</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="deleteGroup">
         <property name="title">
          <string>Delete Data</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Delete by Symbol:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="deleteSymbol">
            <property name="placeholderText">
             <string>Symbol to Delete</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deleteButton">
            <property name="text">
             <string>Delete</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="statsGroup">
         <property name="title">
          <string>Sort and Statistics</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QPushButton" name="sortButton">
            <property name="text">
             <string>Sort by Price (Ascending)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Statistics Function:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="statsCombo">
            <item>
             <property name="text">
              <string>mean</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>sum</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>count</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>min</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>max</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="statsButton">
            <property name="text">
             <string>Calculate Group Statistics</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTableWidget" name="tableWidget"/>
    </item>
    <item>
     <widget class="QGroupBox" name="chartGroup">
      <property name="title">
       <string>Charts</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QWidget" name="chartWidget" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>200</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="chartButton">
         <property name="text">
          <string>Generate Charts</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>